eth/downloader/downloader.go

```diff
@@ -1300,6 +1306,12 @@ func (d *Downloader) fetchParts(deliveryCh chan dataPack, deliver func(dataPack)
 			}
 			// Check for fetch request timeouts and demote the responsible peers
 			for pid, fails := range expire() {
+				select {
```

Comment:
Is this needed? The loop is not doing any heavy lifting, and will probably execute almost instantly